biosignals.py

```diff
@@ -0,0 +1,74 @@
+import platform
+import sys
+from dataclasses import dataclass, field
+
+import plux
+
+osDic = {"Darwin": "MacOS",
+         "Linux": "Linux64",
+         "Windows": ("Win32_37", "Win64_37")}
+if platform.system() != "Windows":
+    sys.path.append("PLUX-API-Python3/{}/plux.so".format(osDic[platform.system()]))
+else:
+    if platform.architecture()[0] == '64bit':
+        sys.path.append("PLUX-API-Python3/Win64_37")
+    else:
+        sys.path.append("PLUX-API-Python3/Win32_37")
+
+
+@dataclass
+class Device:
+    """
+    Stores the integer corresponding to the port occupied by the device.
+    Defaults to port 1
+    """
+
+    source: str
+
+    def set_port(self, source: int):
+        self.source = source
+
+
+@dataclass
+class Session(Device):
+    """
+    Stores session data in a dict
+    """
+    Device: Device
+    Session_Data: dict
+    sampling_frequency: int = 1200
+    Resolution: int = 16
+    Allowed_Resolutions: list = field(default_factory=lambda: [8, 16])
+    Allowed_Settings: list = field(default_factory=lambda: ['fs', 'res', 'source'])
+
+    def prepare_dict(self):
+        """
+        Initializes the dictionary where the session's settings
```

Comment:
Again, really happy to see you using doc blocks. The type I use (there are a few) is the [Google Style Guide](https://google.github.io/styleguide/pyguide.html) 

Adding `Args:`, `Returns:` and even `Raises:` is your next step. The idea of this doc block (or doc string) is to tell the developer who may have never seen this method before what this method is doing, what it requires, what it returns and what it may throw.

A lot of popular IDE's nowadays can display this information to the developer without them ever having to open the file to see, so for example if you see in main.py the call `session.prepare_dict()` the developer can usually hold control and hover over it and see everything in a little pop up.

Final words on future learning would be to hint your return types so `def prepare_dict(self) -> None:`

Reply:
again, I need a little clarification on how to properly setup my doc blocks.

Reply:
It's completely up to you, there are quite a few different style guides, you are free to pick any one you wish. Even just a simple `""" This is a brief description of this method """` is better than no doc block. 

As you start to collaborate on more projects with people, or take over projects from other people, it's always good to stick to the style set forward already in the project to stay consistent. (This is true of naming conventions, tabs or spaces etc, a generic style guide)

You can read about the different Python doc string (or as I say doc block) styles here: https://realpython.com/documenting-python-code/#docstring-formats